hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncClientPauseForServerOverloaded.java

```diff
@@ -56,31 +58,44 @@
 import org.apache.hbase.thirdparty.com.google.protobuf.Descriptors.MethodDescriptor;
 
 @Category({ MediumTests.class, ClientTests.class })
-public class TestAsyncClientPauseForCallQueueTooBig {
+public class TestAsyncClientPauseForServerOverloaded {
```

Comment:
Nice test improvements, checking both failure classes. Please also add coverage to check that when the client provides the old value, it is honored (and perhaps also a warning is produced... the `Configuration.addDeprecation` may log the warning for you, I don't remember.)

Reply:
I do think a warning is produced, but I'm not sure how to check for that. However, I am adding a test which will both ensure that the deprecated config works and that the deprecation happened.